numba/tests/test_inlining.py

```diff
@@ -181,5 +181,38 @@ def test_impl():
 
         self.assertTrue('b' in var_map)
 
+    @skip_unsupported
+    def test_inline_call_branch_pruning(self):
```

Comment:
I'm not sure what is this testing? I think the function `test_impl` would compile anyway under `njit` and that it doesn't compile with `InlineTestPipeline` is by virtue of a side effect, perhaps this should really check that the final IR is branchless as the inline and prune would lead to that?

Reply:
Yes, `njit` alone works due to Dispatcher machinery, but this is testing the use of inlining after type inference on such a function which is a different code path. I added additional testing to check the IR as well.

Reply:
Apologies, I don't think I was particularly clear, what I was trying to get at is that it's _running_ the inliner after type inference, but not _ensuring_ it did the expected, so thanks for adding the extra test to assert this.